pkg/instancegroups/instancegroups.go

```diff
@@ -21,137 +21,227 @@ import (
 	"os"
 	"time"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"golang.org/x/net/context"
+	compute "google.golang.org/api/compute/v0.beta"
 	"github.com/golang/glog"
 	"github.com/spf13/cobra"
 	"k8s.io/client-go/pkg/api/v1"
 	api "k8s.io/kops/pkg/apis/kops"
 	"k8s.io/kops/pkg/client/simple"
 	"k8s.io/kops/pkg/featureflag"
-	"k8s.io/kops/pkg/resources"
 	"k8s.io/kops/pkg/validation"
 	"k8s.io/kops/upup/pkg/fi"
-	"k8s.io/kops/upup/pkg/fi/cloudup/awsup"
 	"k8s.io/kubernetes/pkg/kubectl/cmd"
 	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kops/upup/pkg/fi/cloudup/awsup"
+	"k8s.io/kops/upup/pkg/fi/cloudup/gce"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/aws/aws-sdk-go/aws"
+	"k8s.io/kops/pkg/resources"
 )
 
+// TODO how do we refactor this?  I cannot move the code into awsup.AWSCloud as we
+// TODO get a import cycle not allowed
```

Comment:
@justinsb I actually moved most of `CloudInstanceGroups` stuff into a interface in `fi.Cloud`.  It needed to be a lower level interface that was directly called via cloud.

Seems like a lot of code changes, but it is all copy paste refactoring.